plugins/github/api.ml

```diff
@@ -94,36 +94,68 @@ module Status = struct
 end
 
 module CheckRunStatus = struct
-  type conclusion = [`Failure of string | `Success]
+  type conclusion = [`Failure of string | `Success | `Skipped of string]
   type state = [`Queued | `InProgress | `Completed of conclusion]
+  type action = { label:string; description:string; identifier:string }
   type t = {
       state: state;
-      description: string option;
+      summary: string option;
+      text: string option; 
+      actions: action list;
       url: Uri.t option
     }
 
-  let v ?description ?url state =
-    let description = Option.map (Astring.String.with_range ~len:140) description in (* Max GitHub allows *)
-    { state; description; url }
+  let action ~label ~description ~identifier = {label; description; identifier}
 
-  let state_to_string = function
+  let action_to_json {label; description; identifier} =
+    `Assoc [ ("label", `String label)
+           ; ("description", `String description)
+           ; ("identifier", `String identifier)]
+
+  let v ?text ?summary ?url ?(actions=[]) state =
+    (* A maximum of three actions are accepted by GitHub. *)
+    { state; summary; text; actions; url }
+
+  let state_json ~title ~summary ~text ~status () =
+    let output = [ ("title", `String title)
+                 ; ("summary", `String summary) ] @
+                 (match text with None -> [] | Some x -> ["text", `String x])
+    in
+    [ "status", `String status
+    ; "output", `Assoc output ]
+
+  let json_items { state; text; summary; actions; url } =
+    (match url with None -> [] | Some url -> ["details_url", `String (Uri.to_string url)]) @
+    match state with
     | `Queued -> 
-       ["status", `String "queued"]
+      state_json ~title:"Queued"
+                 ~summary:(Option.value summary ~default:"Queued")
+                 ~status:"queued" ~text ()
+
     | `InProgress -> 
-       ["status", `String "inprogress"]
-    | `Completed `Success -> 
-       ["status", `String "completed"
-       ;"conclusion", `String "success"]
-    | `Completed (`Failure msg) -> 
-       ["status", `String "completed"
-       ;"conclusion", `String "failure"
-       ;"output", `Assoc [("title", `String "Failure")
-                         ;("summary", `String msg)]]
+      state_json ~title:"InProgress" 
+                 ~summary:(Option.value summary ~default:"InProgress")
+                 ~status:"inprogress" ~text ()
 
-  let json_items { state; description; url } =
-    state_to_string state @
-    (match description with None -> [] | Some x -> ["description", `String x]) @
-    (match url with None -> [] | Some x -> ["details_url", `String (Uri.to_string x)])
+    | `Completed `Success -> 
+      ("conclusion", `String "success") ::
+      state_json ~title:"Completed" 
+                 ~summary:(Option.value summary ~default:"Completed")
+                 ~status:"completed" ~text ()
```

Comment:
This is OK, but note that doing it this way prevents using automatic indentation (e.g. with ocp-indent). The original style I suggested avoids that problem:
```ocaml
      state_json ()
        ~title:"Queued"
        ~summary:(Option.value summary ~default:"Queued")
        ~status:"queued"
        ~text
```